src/NuGet.Core/NuGet.Packaging/Signing/Timestamp/Rfc3161TimestampProvider.cs

```diff
@@ -46,32 +47,32 @@ public Rfc3161TimestampProvider(Uri timeStampServerUrl)
             _timestamperUrl = timeStampServerUrl ?? throw new ArgumentNullException(nameof(timeStampServerUrl));
         }
 
-#if IS_DESKTOP
+#if IS_SIGNING_SUPPORTED
 
         /// <summary>
         /// Timestamps data present in the TimestampRequest.
         /// </summary>
-        public Task<PrimarySignature> TimestampSignatureAsync(PrimarySignature primarySignature, TimestampRequest request, ILogger logger, CancellationToken token)
+        public async Task<PrimarySignature> TimestampSignatureAsync(PrimarySignature primarySignature, TimestampRequest request, ILogger logger, CancellationToken token)
         {
-            var timestampCms = GetTimestamp(request, logger, token);
-            using (var signatureNativeCms = NativeCms.Decode(primarySignature.GetBytes()))
+            SignedCms timestampCms = await GetTimestampAsync(request, logger, token);
+            using (ICms signatureCms = CmsFactory.Create(primarySignature.GetBytes()))
             {
                 if (request.Target == SignaturePlacement.Countersignature)
                 {
-                    signatureNativeCms.AddTimestampToRepositoryCountersignature(timestampCms);
+                    signatureCms.AddTimestampToRepositoryCountersignature(timestampCms);
                 }
                 else
                 {
-                    signatureNativeCms.AddTimestamp(timestampCms);
+                    signatureCms.AddTimestamp(timestampCms);
                 }
-                return Task.FromResult(PrimarySignature.Load(signatureNativeCms.Encode()));
+                return PrimarySignature.Load(signatureCms.Encode());
             }
         }
 
         /// <summary>
         /// Timestamps data present in the TimestampRequest.
         /// </summary>
-        public SignedCms GetTimestamp(TimestampRequest request, ILogger logger, CancellationToken token)
+        public async Task<SignedCms> GetTimestampAsync(TimestampRequest request, ILogger logger, CancellationToken token)
```

Comment:
This is a breaking API change.  If we're going to break the API, can we make it `internal`, rather than `public`?  It doesn't seem like an API that people using `NuGet.Packaging` in their own apps would care about.

Reply:
I'm afraid we have to break this API.
There is no sync method for net5.0 code path(https://docs.microsoft.com/en-us/dotnet/api/system.security.cryptography.pkcs.rfc3161timestamprequest?view=net-5.0) to send the timestamp request and return the timestamp.
So I have to use HttpClient.PostAsync to get the timestamp. That's why it's async.
I changed it into internal. 
Thanks!